### PR TITLE
Add cid field to object, application, behavior, region and region-manager

### PR DESCRIPTION
**Pull request by SauloSilva**
ref #2059

### CODE REVIEW

File: test/unit/application.spec.js [hunk 1]
```diff
@@ -30,5 +30,15 @@ describe('marionette application', function() {
...
       expect(this.startStub).to.have.been.calledOnce.and.calledWith(this.fooOptions);
     });
+
+    it('should contains cidPrefix', function() {
+      expect(this.app.cidPrefix).to.equal('mna');
+    });
+
+    it('should contains cid on behavior', function() {
+      expect(this.app.cid).to.exist;
+      expect(this.app.cid).to.be.a('string');
```
Comment (jasonLaster) [inline test/unit/application.spec.js]: The and method would be great here. I don't think is a string is needed

Comment (SauloSilva) [inline test/unit/application.spec.js]: @jasonLaster I removed

File: test/unit/application.spec.js [hunk 2]
```diff
@@ -30,5 +30,13 @@ describe('marionette application', function() {
     it('should pass the startup option to the callback', function() {
       expect(this.startStub).to.have.been.calledOnce.and.calledWith(this.fooOptions);
     });
+
+    it('should contains cidPrefix', function() {
```
Comment (ahumphreys87) [inline test/unit/application.spec.js]: could this be reworded? `should have a cidPrefix`

File: test/unit/application.spec.js [hunk 3]
```diff
@@ -30,5 +30,13 @@ describe('marionette application', function() {
...
+      expect(this.app.cidPrefix).to.equal('mna');
+    });
+
+    it('should exist cid', function() {
```
Comment (ahumphreys87) [inline test/unit/application.spec.js]: `should have a cid`

File: test/unit/behaviors.spec.js [hunk 4]
```diff
@@ -151,6 +151,17 @@ describe('Behaviors', function() {
       this.view = new this.View();
     });
 
+    it('should contains cidPrefix', function() {
```
Comment (ahumphreys87) [inline test/unit/behaviors.spec.js]: `should have a cidPrefix`

File: test/unit/behaviors.spec.js [hunk 5]
```diff
@@ -151,6 +151,17 @@ describe('Behaviors', function() {
...
+      var fooBehavior = new this.behaviors.foo();
+      expect(fooBehavior.cidPrefix).to.equal('mnb');
+    });
+
+    it('should exist cid', function() {
```
Comment (ahumphreys87) [inline test/unit/behaviors.spec.js]: `should have a cid`

File: test/unit/object.spec.js [hunk 6]
```diff
@@ -43,6 +43,14 @@ describe('marionette object', function() {
     it('should maintain a reference to the options', function() {
       expect(this.object.options).to.deep.equal(this.options);
     });
+
+    it('should contains cidPrefix', function() {
```
Comment (ahumphreys87) [inline test/unit/object.spec.js]: `should have a cidPrefix`

File: test/unit/object.spec.js [hunk 7]
```diff
@@ -43,6 +43,14 @@ describe('marionette object', function() {
...
+      expect(this.object.cidPrefix).to.equal('mno');
+    });
+
+    it('should exist cid', function() {
```
Comment (ahumphreys87) [inline test/unit/object.spec.js]: `should have a cid`

File: test/unit/region-manager.spec.js [hunk 8]
```diff
@@ -27,6 +27,14 @@ describe('regionManager', function() {
       .been.calledWith(this.regions)
       .and.to.have.been.calledOn(this.regionManager);
     });
+
+    it('should contains cidPrefix', function() {
```
Comment (ahumphreys87) [inline test/unit/region-manager.spec.js]: `should have a cidPrefix`

File: test/unit/region-manager.spec.js [hunk 9]
```diff
@@ -27,6 +27,14 @@ describe('regionManager', function() {
...
+      expect(this.regionManager.cidPrefix).to.equal('mnrm');
+    });
+
+    it('should exist cid', function() {
```
Comment (ahumphreys87) [inline test/unit/region-manager.spec.js]: `should have a cid`

File: test/unit/setup/helpers.js [hunk 10]
```diff
@@ -15,6 +15,14 @@ var clearFixtures = function () {
   $body.empty();
 };
 
+before(function() {
+  this.checkProperties = function(block, blacklist) {
+    blacklist = blacklist ? blacklist.push('cid') : 'cid'
+    var props = _.partial(_.omit, _, blacklist);
+    block.call(this, props);
+  }
```
Comment (ahumphreys87) [inline test/unit/setup/helpers.js]: needs a semi-colon, not sure why jscs didn't pick this up :confused:

**Comment by ahumphreys87** [comment]
@SauloSilva sorry for all the comments, its basically just 1 thing though. This is great work - almost there!

File: test/unit/setup/helpers.js (hunk 10 above)
Comment (SauloSilva) [inline test/unit/setup/helpers.js]: Was missing require directory three for jscs and jshint. Look this lines [jscs src](https://github.com/marionettejs/backbone.marionette/pull/2524/files#diff-35b4a816e0441e6a375cd925af50752cR207) and [jslint](https://github.com/marionettejs/backbone.marionette/pull/2524/files#diff-35b4a816e0441e6a375cd925af50752cR189). Thanks

**Comment by SauloSilva** [comment]
@ahumphreys87 no problems, thanks for hints. We are almost there! :D

**Comment by jasonLaster** [comment]
:thumbsup:

**Comment by ahumphreys87** [comment]
:+1:

**Pull request merged**